src/software/thunderscope/thunderscope_main.py

```diff
@@ -197,55 +207,82 @@
-    # the provided multicast channel.
-    if args.run_blue:
-
-        tscope = Thunderscope(
-            layout_path=args.layout,
-            load_blue=True,
-            load_yellow=False,
-            load_diagnostics=True,
-            load_gamecontroller=False,
-            visualization_buffer_size=args.visualization_buffer_size,
-        )
-
-        proto_unix_io = tscope.blue_full_system_proto_unix_io
-        runtime_dir = args.blue_full_system_runtime_dir
-        friendly_colour_yellow = False
-        debug = args.debug_blue_full_system
-    elif args.run_yellow:
+    # When we are running with real robots. Possible Running Options:
+    #
+    #       Run Blue
+    #       Run Yellow
+    #       Run Blue + Diagnostics
+    #       Run Yellow + Diagnostics
+    #       Run Diagnostics
+    #
+    # We want to run either 1 instance of AI or 1 instance of RobotCommunication or both which will
+    # send/recv packets over the provided multicast channel.
 
+    if args.run_blue or args.run_yellow or args.run_diagnostics:
         tscope = Thunderscope(
             layout_path=args.layout,
-            load_blue=False,
-            load_yellow=True,
-            load_diagnostics=True,
+            load_blue=bool(args.run_blue),
+            load_yellow=bool(args.run_yellow),
+            load_diagnostics=bool(args.run_diagnostics),
             load_gamecontroller=False,
             visualization_buffer_size=args.visualization_buffer_size,
         )
 
-        proto_unix_io = tscope.yellow_full_system_proto_unix_io
-        runtime_dir = args.yellow_full_system_runtime_dir
-        friendly_colour_yellow = True
-        debug = args.debug_yellow_full_system
+        current_proto_unix_io = None
+        current_mode = RobotCommunicationMode.NONE
+
+        if args.run_blue:
+            current_proto_unix_io = tscope.blue_full_system_proto_unix_io
+            runtime_dir = args.blue_full_system_runtime_dir
+            friendly_colour_yellow = False
+            debug = args.debug_blue_full_system
+        elif args.run_yellow:
+            current_proto_unix_io = tscope.yellow_full_system_proto_unix_io
+            runtime_dir = args.yellow_full_system_runtime_dir
+            friendly_colour_yellow = True
+            debug = args.debug_yellow_full_system
+
+        # if either fullsystem is running, mode is fullsystem
+        if args.run_blue or args.run_yellow:
+            current_mode = RobotCommunicationMode.FULLSYSTEM
+
+        # this proto will be the same as the fullsystem one if fullsystem is enabled
+        if args.run_diagnostics:
+            current_proto_unix_io = tscope.robot_diagnostics_proto_unix_io
+
+            # switches to both mode if fullsystem enabled, or just diagnostics if not
+            current_mode = (
+                RobotCommunicationMode.BOTH
+                if current_mode == RobotCommunicationMode.FULLSYSTEM
+                else RobotCommunicationMode.DIAGNOSTICS
+            )
```

Comment:
in the case that it's both, does it matter if current_proto_unix_io is robot_diagnostics_proto_unix_io or full_system?

Reply:
`robot_diagnostics_proto_unix_io` should be the same as the one used by fullsystem in the case of `BOTH`. We should only have one proto unix io no matter the mode we're running in.
https://github.com/UBC-Thunderbots/Software/blob/84e41a9f9ad62ee2265475da5b9886d145693841/src/software/thunderscope/thunderscope.py#L181-L191